tests/30rooms/70publicroomslist.pl

```diff
@@ -0,0 +1,85 @@
+test "Name/topic keys are corrct",
+   requires => [ $main::API_CLIENTS[0], local_user_fixture() ],
+
+   check => sub {
+      my ( $http, $user ) = @_;
+
+      my %rooms = (
+         publicroomalias_no_name => {},
+         publicroomalias_with_name => {
+            name => "name_1",
+         },
+         publicroomalias_with_topic => {
+            topic => "topic_1",
+         },
+         publicroomalias_with_name_topic => {
+            name => "name_2",
+            topic => "topic_2",
+         },
+      );
+
+      Future->needs_all( map {
+         my $alias_local = $_;
+         my $room = %rooms->{$alias_local};
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => $alias_local,
+            name => $room->{name},
+            topic => $room->{topic},
+         )
+      } keys %rooms )
+      ->then( sub {
+         $http->do_request_json(
+            method => "GET",
+            uri    => "/r0/publicRooms",
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         log_if_fail "publicRooms", $body;
+
+         assert_json_keys( $body, qw( start end chunk ));
+         assert_json_list( $body->{chunk} );
+
+         my %seen = map {
+            $_ => 0,
+         } keys ( %rooms );
+
+         foreach my $room ( @{ $body->{chunk} } ) {
+            assert_json_keys( $room,
+               qw( world_readable guest_can_join num_joined_members aliases )
+            );
+
+            my $name = $room->{name};
+            my $topic = $room->{topic};
+            my $canonical_alias = $room->{canonical_alias};
+
+            my $aliases = $room->{aliases};
+            foreach my $alias ( @{$aliases} ) {
+               foreach my $alias_local ( keys %rooms ) {
+                  if ( $alias =~ m/^\Q#$alias_local:\E/ ) {
```

Comment:
Rather than have the entire body of the `for` loop contained inside another `if`, it would avoid an additional layer of indentation to do

    $alias =~ m/^\Q#$alias_local:\E/ or next;